### DISCUSSION TITLE
Inserting throats to X-ray image

### DISCUSSION THREAD

**Discussion by dpereponov** [PMEAL/porespy#566] [Q&A] [answered]
Hello! 

Currently, I'm working on the creation of the microfluidic chip using a scan of computer tomography. 
The resulting CT image shows only large pores without any connectivity. Using the SNOW algorithm, I found the centers of all the pores. If it's possible, could you tell me which function (or a set of them) will help me add pore throats?

I really appreciate any help you can provide.

**Comment by jgostick** [comment, marked as answer]
So your CT scan did not resolve the throats?  I wonder if you could use just a regular 2D microscope image since it's a microfluidic chip?  We had a [discussion about this with someone else recently](https://github.com/PMEAL/porespy/discussions/548).  Anyway, assuming that you must proceed with your CT image, the first option that occurs to me is a Delaunay tessellation.  If you have points at the centers of each pore, you can use the Delaunay tessellation to connect nearby pores.  OpenPNM has a function for converting the results of a Delaunay tessellation into a pore network (``tri_to_am``).  It's not totally straight-forward to go from the the ``am`` to a useful pore network, so if you go this route I can help.  Perhaps you can start by posting your image here, or sending to me at jgostick at gmail.com?

**Reply by dpereponov** [reply]
Thank you, Professor! I have sent you the image by email.
